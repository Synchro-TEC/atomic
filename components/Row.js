import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";

const Row = ({ className, style, withGutter, children, ...otherProps }) => {
  const baseClass = withGutter ? "sv-row--with-gutter" : "sv-row";
  const rootClass = cx(baseClass, { withGutter }, className);

  return (
    <div className={rootClass} style={style} {...otherProps}>
      {children}
    </div>
  );
};

Row.propTypes = {
  className: PropTypes.string,
  withGutter: PropTypes.bool,
  style: PropTypes.object,
};

export default Row;
