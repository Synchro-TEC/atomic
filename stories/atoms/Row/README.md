The `<Row>` and `<Col>` component.