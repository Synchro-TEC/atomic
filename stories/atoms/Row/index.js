import React from 'react';
import { text, boolean, object } from '@storybook/addon-knobs';

import readme from './README.md';
import Row from '../../../components/atoms/Row/Row';
import Col from '../../../components/atoms/Col/Col';

const component = () => (
  <div>
    <Row
      withGutter={boolean('withGutter', true)}
      className={text('ClassName', 'sv-bg-color--blue-50')}
      style={object('Style', {})}
    >
      <Col className="sv-bg-color--blue-300">col-1</Col>
      <Col className="sv-bg-color--green-300">col-2</Col>
    </Row>
    <Row
      withGutter={boolean('withGutter', true)}
      className={text('ClassName', 'sv-bg-color--blue-50')}
      style={object('Style', {})}
    >
      <Col className="sv-bg-color--blue-300">col-1</Col>
      <Col className="sv-bg-color--green-300">col-2</Col>
      <Col className="sv-bg-color--yellow-300">col-3</Col>
    </Row>
    <Row withGutter={boolean('withGutter', true)} className="sv-bg-color--blue-50 sv-pa--10">
      <Col className="sv-bg-color--blue-300">col-1</Col>
      <Col className="sv-bg-color--green-300">col-2</Col>
      <Col className="sv-bg-color--yellow-300">col-3</Col>
    </Row>
  </div>
);

export default [readme, component];
