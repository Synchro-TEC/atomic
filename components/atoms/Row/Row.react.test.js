import React from 'react';
import Row from './Row';

describe('Row Component', () => {
  test('Render Row Component', () => {
    const row = shallow(<Row />);
    expect(row).toMatchSnapshot();
  });
});
