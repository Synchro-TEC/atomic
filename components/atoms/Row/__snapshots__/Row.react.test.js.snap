// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component Render Row Component 1`] = `
<div
  className="sv-row"
/>
`;
